sources/packages/backend/libs/test-utils/src/factories/designation-agreement.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  DesignationAgreement,
+  DesignationAgreementLocation,
+  DesignationAgreementStatus,
+  Institution,
+  InstitutionLocation,
+  User,
+} from "@sims/sims-db";
+import * as faker from "faker";
+import { INSTITUTION_TYPE_BC_PRIVATE } from "../constants";
+
+export function createFakeDesignationAgreement(
+  fakeInstitution: Institution,
+  fakeUser: User,
+  fakeInstitutionLocations: InstitutionLocation[],
+  designationStatus?: DesignationAgreementStatus,
+): DesignationAgreement {
+  const now = new Date();
+
+  const fakeDesignationAgreement = new DesignationAgreement();
+  fakeDesignationAgreement.institution = {
```

Comment:
The idea of having the `fake` something is either to receive the object to be used or create a completely fake one. Please see below the idea. Same for others, please see the existing classes.
```ts
fakeDesignationAgreement.institution = fakeInstitution ?? createFakeInstitution()
```